### ISSUE TITLE
[BUG] npm exec permission when PWD has different owner

### ISSUE THREAD

**Issue by miki725** [npm/cli#5533]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### This issue exists in the latest npm version

- [X] I am using the latest npm

### Current Behavior

When running `npm exec` in a folder which has different owner compared to the binary which you are attempting to execute it causes issues. In this case docker image has pre-installed `pyright` binary which will be installed and owned by `root` however at runtime we mount a different folder which has different permissions.

Reproduction with docker:

```Dockerfile
FROM node:18
RUN npm exec -- pyright@1.1.271 --version
```

then run.:

```bash
yay@earth ~> docker build -t npmtest -f Dockerfile.npm .
Sending build context to Docker daemon  3.072kB
Step 1/2 : FROM node:18
 ---> 2577ab2cda97
Step 2/2 : RUN npm exec -- pyright@1.1.271 --version
 ---> Running in 04ac46be5ca9
npm WARN exec The following package was not found and will be installed: pyright@1.1.271
pyright 1.1.271
npm notice
npm notice New patch version of npm available! 8.19.1 -> 8.19.2
npm notice Changelog: <https://github.com/npm/cli/releases/tag/v8.19.2>
npm notice Run `npm install -g npm@8.19.2` to update!
npm notice
Removing intermediate container 04ac46be5ca9
 ---> 565d21f59c07
Successfully built 565d21f59c07
Successfully tagged npmtest:latest


yay@earth ~> docker run -it --rm  -w $PWD -v $PWD:$PWD npmtest bash

root@393c328ca1a1:/home/yay# id
uid=0(root) gid=0(root) groups=0(root)

root@393c328ca1a1:/home/yay# ls -la . | head -n2
total 156
drwx------ 1  970  970    406 Sep 19 17:48 .

root@393c328ca1a1:/home/yay# npm --version
8.19.1

root@393c328ca1a1:/home/yay# npm exec -- pyright@1.1.271 --version
sh: 1: pyright: Permission denied

# -----------------
# but if I cd into folder owned by root it works fine

root@393c328ca1a1:/home/yay# cd

root@393c328ca1a1:~# pwd
/root

root@393c328ca1a1:~# npm exec -- pyright@1.1.271 --version
pyright 1.1.271
```

### Expected Behavior

`PWD` owner should not impact `npm exec` and `npm exec -- pyright@1.1.271 --version` above should succeed in any folder.

### Steps To Reproduce

See above. Small Dockerfile is provided which should be able to reproduce it.

### Environment

```bash
root@3547d7857b7f:/home/yay# cat /etc/*release
PRETTY_NAME="Debian GNU/Linux 11 (bullseye)"
NAME="Debian GNU/Linux"
VERSION_ID="11"
VERSION="11 (bullseye)"
VERSION_CODENAME=bullseye
ID=debian
HOME_URL="https://www.debian.org/"
SUPPORT_URL="https://www.debian.org/support"
BUG_REPORT_URL="https://bugs.debian.org/"

root@3547d7857b7f:/home/yay# node --version
v18.9.0

root@3547d7857b7f:/home/yay# npm --version
8.19.1
```

- npm config:

```ini
root@393c328ca1a1:/home/yay# npm config ls
; node bin location = /usr/local/bin/node
; node version = v18.9.0
; npm local prefix = /home/yay
; npm version = 8.19.1
; cwd = /home/yay
; HOME = /root
; Run `npm config ls -l` to show all defaults.
```

**Comment by wraithgar** [comment]
Can you try this in npm@9.0.0? npm now no longer attempts to discover or change folder permissions.

**Comment by miki725** [comment]
still no luck:

```sh
~> cat Dockerfile.npm
FROM node:18
RUN npm install -g npm@9.0.0
RUN npm exec -- pyright@1.1.271 --version

~> docker build -t npmtest -f Dockerfile.npm .
...
Successfully tagged npmtest:latest

~> docker run -it --rm  -w $PWD -v $PWD:$PWD npmtest bash
root@7aee7008a870:/home/yay# npm --version
9.0.0
root@7aee7008a870:/home/yay# npm exec -- pyright@1.1.271 --version
sh: 1: pyright: Permission denied
root@7aee7008a870:/home/yay# id
uid=0(root) gid=0(root) groups=0(root)
root@7aee7008a870:/home/yay# cd
root@7aee7008a870:~# npm exec -- pyright@1.1.271 --version
pyright 1.1.271
```

**Comment by milaninfy** [comment]
This appears to be fixed on latest npm ( 10.8.3 ) Feel free to reopen if i have missed something or done differently. 
```sh
~/workarea/rep $ docker run -it --rm  -w $PWD -v $PWD:$PWD npmtest bash
root@a1f0e7d9e5fa:/Users/milaninfy/workarea/rep# id
uid=0(root) gid=0(root) groups=0(root)
root@a1f0e7d9e5fa:/Users/milaninfy/workarea/rep# ls -la . | head -n2
total 8
drwxr-xr-x 5 root root  160 Aug 30 20:29 .
root@a1f0e7d9e5fa:/Users/milaninfy/workarea/rep# npm --version
10.7.0
root@a1f0e7d9e5fa:/Users/milaninfy/workarea/rep# pm exec -- pyright@1.1.271 --version
bash: pm: command not found
root@a1f0e7d9e5fa:/Users/milaninfy/workarea/rep# npm exec -- pyright@1.1.271 --version
pyright 1.1.271
root@a1f0e7d9e5fa:/Users/milaninfy/workarea/rep# cd
root@a1f0e7d9e5fa:~# pwd
/root
root@a1f0e7d9e5fa:~# npm exec -- pyright@1.1.271 --version
pyright 1.1.271

```
